tf2/src/buffer_core.cpp

```diff
@@ -528,23 +528,26 @@ tf2::TF2Error BufferCore::walkToTopParent(F& f, TimePoint time, CompactFrameID t
   if (frame_chain)
   {
     // Pruning: Compare the chains starting at the parent (end) until they differ
```

Comment:
Correct me if I'm wrong, but changing the direction of iteration contradicts this comment. I'm not sure of the importance of starting from the parent, but I think it's best to change this back to reflect the original logic.

Reply:
The index was reverted, but the vector is still iterated in the same direction (with the proper arithmetic conversions).

The change is needed to avoid the sign/unsign comparison.

I will split this PR in two, and we can do the more complex changes after the release.

Reply:
@jacobperron can you re-check if the logic is valid?

Reply:
LGTM